_LowLevelCode/CPP/cpp_communicator/cpp_communicator.cpp

```diff
@@ -67,7 +128,14 @@ void mexFunction(int nlhs, mxArray *plhs[], int nrhs, const mxArray *prhs[])
     switch (work_type)
     {
     case(init_mpi): { // Initialize MPI communications and return labIndex and numLabs
-        pCommunicatorHolder->class_ptr->init();
+        pCommunicatorHolder->class_ptr->init(false, AddPar.assynch_queue_length,AddPar.data_message_tag);
+        n_workers = pCommunicatorHolder->class_ptr->numProcs;
```

Comment:
where is this `n_workers` being used?

Reply:
That's the leftover from the previous implementation. Removed